### DISCUSSION TITLE
Base styling in Shadow DOM

### DISCUSSION THREAD

**Discussion by sidharthramesh** [tailwindlabs/tailwindcss#3671] [Help]
As you might know, the Shadow DOM does not usually have `html` and `body` elements. Some of the styling (especially the base and preflight) in TailwindCSS is done on these elements. As a work around I have replaced the `html` styles to the `:host` during the build process using [postcss-replace](https://www.npmjs.com/package/postcss-replace). However, there is still a lot of `body` styles that are not translated into the Shadow DOM. Can anyone help? What is the equivalent of `body` in the Shadow DOM for styling purposes?

**Comment by timomayer** [comment]
i think its more tailwind who should adapt here to be compatible with web components, i would appreciate if web components and shadow dom would be officially supported by tailwind
